services/app-api/src/s3/s3Error.ts

```diff
@@ -0,0 +1,23 @@
+export const S3ErrorCodes = ['NETWORK_ERROR'] as const
+type S3ErrorCode = (typeof S3ErrorCodes)[number] // iterable union type
+
+export type S3Error = {
+    code: S3ErrorCode
+    message: string
+}
+
+// Wow this seems complicated. If there are cleaner ways to do this I'd like to know it.
```

Comment:
We should use a real Error object here, this kind of fake error was something I did wrong early on. Just return actual errors everywhere and you can use `instanceof Error` instead of this function. If we need to get a specific message from the error, we can make an `S3Error` that inherits from Error

Reply:
If it's too much hassle that's ok b/c we are sharing this code with the FE so it's reasonable just to keep it the same while we fix the packaging

Reply:
I fixed everything else but am going to leave this as-is for now to revisit later